xlators/encryption/crypt/src/data.c

```diff
@@ -287,20 +287,28 @@ data_alloc_block(xlator_t *this, crypt_local_t *local, int32_t block_size)
 {
     struct iobuf *iobuf = NULL;
 
+    int is_iobref_new = 0;
+
     iobuf = iobuf_get2(this->ctx->iobuf_pool, block_size);
     if (!iobuf) {
         gf_log("crypt", GF_LOG_ERROR, "Failed to get iobuf");
         return NULL;
     }
     if (!local->iobref_data) {
         local->iobref_data = iobref_new();
+        is_iobref_new = 1;
         if (!local->iobref_data) {
             gf_log("crypt", GF_LOG_ERROR, "Failed to get iobref");
             iobuf_unref(iobuf);
             return NULL;
         }
     }
     iobref_add(local->iobref_data, iobuf);
+
+    if(is_iobref_new == 1){
```

Comment:
if(is_iobref_new) {} is good enough.

Reply:
thanks ! 38add7b33c6f107439bc98b6aa99da6561aabbc5